### PR TITLE
Feature/config file

### PR DESCRIPTION
**Pull request by ajsager**

### CODE REVIEW

File: bin/newman [hunk 1]
```diff
@@ -21,10 +21,25 @@ var color   = require('cli-color'),
...
 
+function parseConfigFile() {
+	var config,
+		option;
+
+	Helpers.validateConfigFile(program.configFile);
+	config = Helpers.validateAndReturnConfig(program.configFile);
+
+	for (option in config) {
+		if (!program[option]) {
```
Comment (shamasis) [inline bin/newman]: `!program[option]` would pass for `false` and zero. Not sure whether commander parses to Boolean and number, but surely checking for own property or inequality with null and undefined is better.

File: bin/newman [hunk 2]
```diff
@@ -55,17 +70,22 @@ function parseArguments() {
 		console.log('  A collection file or URL is mandatory');
 		console.log('  Examples:');
 		console.log('');
+		console.log('    newman -C CONFIG_FILE');
 		console.log('    newman -c POSTMAN_COLLECTION');
 		console.log('    newman -u POSTMAN_COLLECTION -e POSTMAN_ENVIRONMENT');
 		console.log('    newman -c POSTMAN_COLLECTION -f SAMPLE_FOLDER');
 		console.log('');
-		console.log('  For more documentation, issues and examples head over to https://github.com/a85/Newman');
+		console.log('  For more documentation, issues and examples head over to https://github.com/postmanlabs/newman');
```
Comment (shamasis) [inline bin/newman]: Ah. @pjsager nice catch. :+1:

File: src/utilities/Helpers.js [hunk 3]
```diff
@@ -12,6 +12,55 @@ var jsface = require('jsface'),
...
+        try {
+            config = require(file);
+        }
+        catch (RefrenceError) {
+            Errors.terminateWithError("The config file specified is not a valid JSON file");
+        }
+
+        Object.keys(config).forEach(function (option) {
+            if (!validOptions[option] || (typeof config[option] !== validOptions[option])) {
+                Errors.terminateWithError("Config JSON has invalid options");
```
Comment (shamasis) [inline src/utilities/Helpers.js]: Do we want to make the config file this strict? What if we allow these, but merge only the relevant ones?

File: src/utilities/Helpers.js [hunk 4]
```diff
@@ -12,6 +12,55 @@ var jsface = require('jsface'),
...
+                insecure: "boolean",
+                tls: "boolean",
+                encoding: "string",
+                exitCode: "boolean",
+                outputFile: "string",
+                testReportFile: "string",
+                import: "string",
+                pretty: "boolean",
+                html: "string"
+            },
```
Comment (shamasis) [inline src/utilities/Helpers.js]: Now that's one more place to maintain the list of config. Not that it's too much pain, but is there a way to pick this up straight from commander definition? :-p

**Comment by shamasis** [comment]
@ajsager - nicely done. :-)
there are some merge conflicts though.

File: src/utilities/Helpers.js (hunk 3 above)
Comment (czardoz) [inline src/utilities/Helpers.js]: I prefer strict.. There's never unexpected behavior that way, and it forces people to look at exactly what options are supported

**Comment by shamasis** [comment]
I also see that no new tests are added for the config file feature.

**Comment by ajsager** [comment]
These changes were from @elssar.  I mean to make a future change, but have not made any commits yet.

**Comment by shamasis** [comment]
@ajsager - I guess that's why he didn't send a PR. We will wait for @elssar to send a fresh PR then.

**Pull request closed without merging**